aptos-move/aptos-aggregator/src/delayed_change.rs

```diff
@@ -152,17 +153,82 @@ impl<I: Copy + Clone> DelayedChange<I> {
                 Ok(Create(Snapshot(new_data)))
             },
             (Some(Apply(AggregatorDelta { delta: prev_delta })), Apply(SnapshotDelta { delta: next_delta, base_aggregator })) => {
-                let new_delta = DeltaOp::create_merged_delta(prev_delta, next_delta)?;
+                let new_delta = DeltaWithMax::create_merged_delta(prev_delta, next_delta)?;
                 Ok(Apply(SnapshotDelta { delta: new_delta, base_aggregator: *base_aggregator }))
             },
         }
     }
 }
 
+// TODO: See if we need these separate/duplicate classes or not
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum DelayedApplyEntry<I: Clone> {
+    AggregatorDelta {
+        delta: DeltaOp,
```

Comment:
I'm a little bit confused with DelayedApplyChange and DelayedApplyEntry. The only difference seems to be one uses DeltaOp and the other uses DeltaWithMax. 
I thought we are using DeltaOps for Aggregator V1 and DeltaWithMax for Aggregator V2. Isn't that the case?

Reply:
when we call a read(), we go to versioned_delayed_fields to aggregate deltas. at that point - having DeltaHistory is probably useful, instead of giving bogus numbers to the caller.

and so , in executor.rs, after we execute, and get read_set and write_set separately, we can merge it together and store it in versioned_delayed_fields.

This needs some cleanup and refactoring, but is code-correct now, so want to do that separately